crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -38,6 +80,35 @@ pub(crate) struct TransactionBlockEffects {
     // pub checkpoint: Checkpoint
 }
 
+impl From<&SuiTransactionBlockEffects> for TransactionBlockEffects {
+    fn from(tx_effects: &SuiTransactionBlockEffects) -> Self {
+        let (status, errors) = match tx_effects.status() {
+            SuiExecutionStatus::Success => (ExecutionStatus::Success, None),
+            SuiExecutionStatus::Failure { error } => {
+                (ExecutionStatus::Failure, Some(error.clone()))
+            }
+        };
+
+        Self {
+            digest: TransactionDigest::from_array(tx_effects.transaction_digest().into_inner()),
```

Comment:
This is the wrong digest, and unfortunately I believe the JSON RPC doesn't store the effects digest, so you may not be able to access it via this data provider -- so I would suggest leaving this field off for now.

I have left a [comment on the data layer schema](https://github.com/MystenLabs/sui/pull/13574/files#r1324231969) enquiring whether we should support effect digests there, or remove support for them on the GraphQL side.  Let's wait to here what they say.

P.S., as per my comment here: https://github.com/MystenLabs/sui/pull/13660#pullrequestreview-1615050100 we shouldn't have a dedicated scalar for `TransactionDigest`.  It should just be a `String` based on our initial design discussions where we decided to limit our use of custom scalars, to reduce the cost for adopting this schema on the client (each custom scalar must be backed by a custom serializer/deserializer on the client).  In this case it's even adding to the confusion, because this isn't a `TransactionDigest`, it's an `EffectsDigest`.  Fixing this doesn't need to block this PR, but could you handle it in a follow-up @wlmyng ?

Reply:
The way things are leaning so far is towards not having to refer to effects digest at all, so that makes things simpler!  For reference, I've updated the draft schema in https://github.com/MystenLabs/sui/pull/13765, to give an idea of what that means for us (still need to leave some time in case other folks have thoughts to share on this though, before locking that change in).

Reply:
Ah missed your latest comment